tests/Resources/ResourceResetter.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Resources;
+
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Backups and reverts testable resources to its initial state.
+ */
+class ResourceResetter
+{
+    /**
+     * @var Filesystem|null
+     */
+    private $filesystem;
+
+    /**
+     * @var string|null
+     */
+    private $backupRootDir;
+
+    /**
+     * @param Filesystem|null $filesystem
+     * @param string|null $backupRootDir
+     */
+    public function __construct(
+        ?Filesystem $filesystem = null,
+        ?string $backupRootDir = null
+    ) {
+        $this->filesystem = $filesystem ?: new Filesystem();
+        $this->backupRootDir = $backupRootDir ?: sys_get_temp_dir();
+    }
+
+    /**
+     * Name for directory of test images in system tmp dir
+     */
+    const BACKUP_TEST_IMG_DIR = 'ps_backup_test_img';
+    const BACKUP_TEST_DOWNLOADS_DIR = 'ps_backup_test_download';
```

Comment:
```suggestion
    public const BACKUP_TEST_IMG_DIR = 'ps_backup_test_img';
    public const BACKUP_TEST_DOWNLOADS_DIR = 'ps_backup_test_download';
```

Reply:
not sure about this. Const is public by default, but the main issue is that we don't define it this way anywhere else, so I'm not in favor of changing our "current convention"

Reply:
@zuk3975 If we don't define it, they are public. So define it ;)

Reply:
ok sorry my bad, I see it in PSR-2 `Visibility MUST be declared on all constants if your project PHP minimum version supports constant visibilities (PHP 7.1 or later).`

Reply:
Ooooh I did not know that. Thanks I learned something today 👍